R/gh_token.R

```diff
@@ -142,3 +144,30 @@ obfuscate <- function(x, first = 4, last = 4) {
     substr(x, start = nchar(x) - last + 1, stop = nchar(x))
   )
 }
+
+get_connect_session <- function() {
```

Comment:
This is really the Shiny session, when running on Connect, right?

Might be relevant to make that distinction because with the client credentials flow, IIUC it will be possible to use oauth tokens from Connect in non-Shiny contexts (R markdown or otherwise).

Reply:
Yes, although GitHub doesn't support the client credentials flow, so this isn't directly relevant to the `gh` package.

(I'm planning to add a dedicated API to `connectcreds` for service accounts, but it won't consume a Shiny session.)